spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/validation/DefaultValidationService.java

```diff
@@ -78,7 +78,7 @@ public boolean validate(String name, ApplicationType appType) {
 					result = validateDockerResource(dockerValidatorProperties, (DockerResource) resource);
 				}
 				else {
-					new BootClassLoaderFactory(resolveAsArchive(appRegistry.getAppResource(registration)), null)
+					new BootClassLoaderFactory(resolveAsArchive(resource), null)
```

Comment:
This change is removing the redundant code getting the `resource`, but the NPE is in the method `resolveAsArchive` since the `resource` passed into that method can be null, giving the stack trace in the issue.

Reply:
I'll fix the issue as you describe, but please read my comment above, where I addressed my concerns on fixing it.